### PR TITLE
feat(gatsby-plugin-catch-links): Add ability to exclude multiple link patterns

### PR DESCRIPTION
**Pull request by alexanderschau**
Adding possibility to exclude multiple paths in the catch links plugin.
refs to #14835

### CODE REVIEW

File: packages/gatsby-plugin-catch-links/src/catch-links.js [hunk 1]
```diff
@@ -152,10 +152,18 @@ export const routeThroughBrowserOrApp = (
 
   if (hashShouldBeFollowed(origin, destination)) return true
 
+  // Regex to test against exclude patterns
```
Comment (xaverfleer) [inline packages/gatsby-plugin-catch-links/src/catch-links.js:155]: Comment adds no information (compare with next line).

Suggestion: Remove comment.

File: packages/gatsby-plugin-catch-links/src/catch-links.js [hunk 2]
```diff
@@ -152,10 +152,18 @@ export const routeThroughBrowserOrApp = (
...
   if (pluginOptions.excludePattern) {
-    const excludeRegex = new RegExp(pluginOptions.excludePattern)
-    if (excludeRegex.test(destination.pathname)) {
-      return true
+    if (Array.isArray(pluginOptions.excludePattern)) {
+      pluginOptions.excludePattern.forEach(pattern => {
+        const excludeRegex = new RegExp(pattern)
+        if (excludeRegex.test(destination.pathname)) return true
```
Comment (xaverfleer) [inline packages/gatsby-plugin-catch-links/src/catch-links.js:160]: Duplicate code (compare with else branch on lines 162-165)

Have you considered adding a helper function?

File: packages/gatsby-plugin-catch-links/src/catch-links.js [hunk 3]
```diff
@@ -152,10 +152,18 @@ export const routeThroughBrowserOrApp = (
 
   if (hashShouldBeFollowed(origin, destination)) return true
 
+  // Regex to test against exclude patterns
   if (pluginOptions.excludePattern) {
-    const excludeRegex = new RegExp(pluginOptions.excludePattern)
-    if (excludeRegex.test(destination.pathname)) {
-      return true
+    if (Array.isArray(pluginOptions.excludePattern)) {
```
Comment (xaverfleer) [inline packages/gatsby-plugin-catch-links/src/catch-links.js:157]: Have you considered to add a variable

```javascript
const patterns = Array.isArray(pluginOptions.excludePattern) ? pluginOptions.excludePattern : [pluginOptions.excludePattern]
```
?

This makes the if/else statement obsolete.

**Review by xaverfleer — COMMENTED** [review]
Looks good to me.

Before merging
- [ ] check comments
- [ ] consider adding automatic tests
- [ ] fix automatic tests (some are failing)

**Comment by j0sh77** [comment]
Hey! Thanks so much for opening this pull request!

Sadly this PR got stale and didn't have any activity for some time. We're trying to do better with PR reviews! To get a better overview of all actionable PRs we're going through all open PRs and triage them. Since we won't be able to do everything and adding new features _always_ means added maintenance burden, we have to be more picky about what's beneficial for the average user and the project itself longterm.

Unfortunately we won't merge this PR at the moment because:
- Original tests are failing
- There hasn't been much activity or expressed need for it

We absolutely want to have you as a contributor and are sorry for any inconveniences we caused with replying too late to this PR.

Thanks for submitting to Gatsby! :purple_heart:

**Pull request closed without merging**